tests/integration/devfile/cmd_devfile_deploy_test.go

```diff
@@ -71,8 +72,59 @@ var _ = Describe("odo devfile deploy command tests", func() {
 		It("Should error out with 'dockerfile required for build.'", func() {
 			helper.CmdShouldPass("odo", "create", "nodejs", "--project", namespace, cmpName)
 			helper.CmdShouldPass("odo", "url", "create", "--port", "3000")
-			cmdOutput := helper.CmdShouldFail("odo", "deploy", "--tag", imageTag, "--devfile", "devfile.yaml")
+			cmdOutput := helper.CmdShouldFail("odo", "deploy", "--tag", imageTag)
 			Expect(cmdOutput).To(ContainSubstring("dockerfile required for build. No 'dockerfile' field found in devfile, or Dockerfile found in project directory"))
 		})
 	})
+
+	Context("Verify error when no manifest definition exists in devfile", func() {
+		It("Should error out with 'Unable to deploy as alpha.deployment-manifest is not defined in devfile.yaml'", func() {
+			helper.CmdShouldPass("odo", "create", "nodejs", "--project", namespace, cmpName)
+			helper.CmdShouldPass("odo", "url", "create", "--port", "3000")
+			helper.CopyExampleDevFile(filepath.Join("source", "devfilesV2", "nodejs", "devfile-no-manifest.yaml"), filepath.Join(context, "devfile.yaml"))
+
+			cmdOutput := helper.CmdShouldFail("odo", "deploy", "--tag", imageTag, "--deployOnly")
+			Expect(cmdOutput).To(ContainSubstring("Unable to deploy as alpha.deployment-manifest is not defined in devfile.yaml"))
+		})
+	})
+
+	Context("Verify error when invalid manifest definition exists in devfile", func() {
+		It("Should error out with 'Invalid manifest url'", func() {
+			helper.CmdShouldPass("odo", "create", "nodejs", "--project", namespace, cmpName)
+			helper.CmdShouldPass("odo", "url", "create", "--port", "3000")
+			helper.CopyExampleDevFile(filepath.Join("source", "devfilesV2", "nodejs", "devfile.yaml"), filepath.Join(context, "devfile.yaml"))
+			err := helper.ReplaceDevfileField("devfile.yaml", "alpha.deployment-manifest", "google.com")
+			Expect(err).To(BeNil())
+
+			cmdOutput := helper.CmdShouldFail("odo", "deploy", "--tag", imageTag, "--deployOnly")
+			Expect(cmdOutput).To(ContainSubstring("Invalid manifest url"))
+		})
+	})
+
+	Context("Verify error when manifest file doesnt exist on web", func() {
+		It("Should error out with 'Unable to download manifest'", func() {
+			helper.CmdShouldPass("odo", "create", "nodejs", "--project", namespace, cmpName)
+			helper.CmdShouldPass("odo", "url", "create", "--port", "3000")
+			helper.CopyExampleDevFile(filepath.Join("source", "devfilesV2", "nodejs", "devfile.yaml"), filepath.Join(context, "devfile.yaml"))
+			err := helper.ReplaceDevfileField("devfile.yaml", "alpha.deployment-manifest", "http://github.com/myfile.yaml")
+			Expect(err).To(BeNil())
+
+			cmdOutput := helper.CmdShouldFail("odo", "deploy", "--tag", imageTag, "--deployOnly")
```

Comment:
We need to remove this `deployOnly` since we haven't discussed it in the proposal

Reply:
Removed from tests. New PR will be raised to remove from actual ODO code.